### PR TITLE
refactor strategy allocation

### PR DESCRIPTION
**Pull request by rolandpo**
refactor strategy allocation with the structured analytics endpoint, replacing #45, prevent some error edge cases

requires removing COMP from analytics if possible otherwise can edit just frontend to exclude it

### CODE REVIEW

File: src/components/Allocation.js [hunk 1]
```diff
@@ -62,66 +76,53 @@ const Allocation = ({ strategies }) => {
...
-                      if (
-                        strategy.name === 'Vault' ||
-                        strategy.name === 'OUSD MetaStrategy' ||
-                        strategy.name === 'Morpho Aave' ||
-                        strategy.name === 'Convex LUSD+3Crv'
-                      )
-                        return
-strategy.name === 'Morpho Compound' && console.log(strategy)
+                    {protocolsSorted?.map((protocol, i) => {
+                      if (protocol.name == 'undefined' || protocol.name === 'Vault') return
```
Comment (shahthepro) [inline src/components/Allocation.js:80]: let's not use the name. You can pass the UUID as key or something and use that here

File: src/components/Allocation.js [hunk 2]
```diff
@@ -62,66 +76,53 @@ const Allocation = ({ strategies }) => {
...
                         <div
                           className="strategy rounded-xl border-2 p-[16px] md:p-8 my-[6px] md:my-[8px]"
-                          key={strategy.name}
+                          key={i}
                           onClick={(e) => {
                             e.preventDefault()
                             setOpen({
                               ...open,
-                              [strategy.name]: !open[strategy.name],
+                              [protocol.name]: !open[protocol.name],
```
Comment (shahthepro) [inline src/components/Allocation.js:89]: Let's use UUID instead of name

File: src/components/Allocation.js [hunk 3]
```diff
@@ -62,66 +76,53 @@ const Allocation = ({ strategies }) => {
...
                           <div>
                             <div className="flex flex-row justify-between">
                               <div className='relative w-1/3 md:w-1/3 lg:w-1/4'>
                                 <Image
-                                  src={assetRootPath(
-                                    `/images/${strategy.name
-                                      .replace(/\s+/g, '-')
-                                      .toLowerCase()}.svg`
-                                  )}
+                                  src={protocolMapping[protocol.name]?.image}
```
Comment (shahthepro) [inline src/components/Allocation.js:97]: Same comment here about name usage

**Review by shahthepro — CHANGES_REQUESTED** [review]
I still see some parts using the name instead of UUID

File: src/components/Allocation.js (hunk 1 above)
Comment (rolandpo) [reply src/components/Allocation.js:80]: the UUID refers to the strategies as defined in analytics, doesn't it? I need to iterate just through protocols here, so Convex, Morpho, Aave, Compound

**Review by sparrowDom — APPROVED** [review]
LGTM

**Review by shahthepro — APPROVED** [review]

**Pull request merged**